src/java/com/puppetlabs/puppetserver/MetricsPuppetProfiler.java

```diff
@@ -54,29 +62,107 @@ public Set<String> getAllMetricIds() {
     }
 
     public Map<String, Timer> getFunctionTimers() {
-        return getTimers(FUNCTION_PATTERN);
+        return this.function_timers;
     }
 
     public Map<String, Timer> getResourceTimers() {
-        return getTimers(RESOURCE_PATTERN);
+        return this.resource_timers;
     }
 
     public Map<String, Timer> getCatalogTimers() {
-        return getTimers(CATALOG_PATTERN);
+        return this.catalog_timers;
     }
 
     public Map<String, Timer> getInliningTimers() {
-        return getTimers(INLINING_PATTERN);
+        return this.inlining_timers;
     }
 
     public Map<String, Timer> getPuppetDBTimers() {
-        return getTimers(PUPPETDB_PATTERN);
+        return this.puppetdb_timers;
     }
 
     @Override
     public void shutdown() {
     }
 
+    private List<String> sliceOfArrayToList(String[] idSegments, int lengthOfID) {
```

Comment:
I cant see how `subList` does not work for this? 

Seems like for example the `getMetricName` could be called with the result of a `metricId.subList(0,2)` as the result of `subList` I think would be mutable?

Reply:
We start with a `String[]` which doesn't have the `subList()` method defined, but we can turn an array into a List with `Arrays.asList()`.

I originally tried `Arrays.asList(ids).subList(0,2)`. But the problem is `asList()` returns a fixed sized List and `subList()` simply returns a view of the original fixed size list. So we end up erring when `getMetricName` attempts to prepend the hostname to the list here: https://github.com/puppetlabs/puppetserver/pull/2880/files#diff-300731d432bcbd4678957c962c2aa20d068698fff7a4441434c3d1ac5e55fdfdR196

I should probably clean up `getMetricName()` but I didn't want to lead with a larger refactor than was strictly necessary (at least after I had changed `getTimers`)

Reply:
Ah, i see, i misread that parameter type.